### DISCUSSION TITLE
post/2021/2021-09-comparison-of-cf-and-ga-data/

### DISCUSSION THREAD

**Discussion by giscus** [zyrikby/website_related#20] [Announcements]
# post/2021/2021-09-comparison-of-cf-and-ga-data/

Some time ago, I wrote an article showing how much data is missed if you rely on only client-side web analytics numbers. To maintain this previous blog post in an actual state, I planned to update it from time to time. However, as you might expect, pretty soon, I got bored collecting manually Cloudflare and Google Analytics data and inserting it into a spreadsheet. As a normal human, I have decided to automate the process. Luckily, both systems provide the APIs that you can use to query and download the data. However, during the development of the data collection script, I discovered several limitations that inclined me to write a new blog post instead of updating the old one.

https://zhauniarovich.com/post/2021/2021-09-comparison-of-cf-and-ga-data/

**Comment by rmminusrslash** [comment]
Google Analytics removes bot traffic. Bot traffic is a significant share daily. Can you see the user agents of the cloudflare logs?

**Reply by zyrikby** [reply]
You're right, this could be just another factor. For instance, I see the days where there are huge spikes in CF data, but GA data produces almost flat traffic. However, I do not think that this is only one factor because the correlation between the data is really high, so this means that for some reason GA records only a fraction of all visits. Unfortunately, with a free CF account you cannot get user-agents information to check this assumption.  

In any case, I am not a SEO professional, I just collect the metrics for myself to see the progress. So as soon as I see the rise for any of these sources, I am happy :)

**Reply by rmminusrslash** [reply]
haha, ok, then I wont tell you more, you can pick the higher number:)

**Reply by zyrikby** [reply]
For the external audience, I report the numbers from both sources. Now, you can pick whatever number you like more )))

**Comment by ItsDanielHarris** [comment]
I have my own PHP code that records visitor information into a database and filters out bots. It works pretty well. This may be more accurate than using Google or Cloudflare.

**Reply by zyrikby** [reply]
Yes, this could be. However, here I do not have a backend, my website is simply a set of static webpages, therefore, I cannot that easily write a backend code to analyze the results.

In any case, I would love to see the results of your analysis. If you would make an experiment, compare the numbers and write a post - let me know.
